File: python/account/current.py

from python.account.account import Account


class Current(Account):
    """
    A concrete implementation of Account for Current account.
    """
    
    def deposit(self, amount):
        """
        To deposit a specified amount into the current account
        """
        self.balance += amount

    def withdraw(self, amount):
        """
        To withdraw a specified amount from the current account
        """
        if self.balance >= amount:
            self.balance -= amount
        else:
            raise ValueError("Insufficient funds")

    def get_balance(self):
        """
        Method to get the current account balance
        """
        return self.balance

    def get_account_number(self):
        """
        Method to get the current account number
        """
        return self.account_number

File: java/src/main/java/org/nitk/hackathon/account/Current.java
package org.nitk.hackathon.account;

public class Current implements Account {
    private String accountNumber;
    private double balance;

    public Current(String accountNumber) {
        this.accountNumber = accountNumber;
        this.balance = 0;
    }

    @Override
    public String getAccountNumber() {
        return accountNumber;
    }

    @Override
    public void deposit(double amount) {
        balance += amount;
    }

    @Override
    public void withdraw(double amount) {
        balance -= amount;
    }

    @Override
    public double getBalance() {
        return balance;
    }

}